src/lib/autocomplete/autocomplete-trigger.ts

```diff
@@ -133,6 +137,16 @@ export class MdAutocompleteTrigger implements ControlValueAccessor, OnDestroy {
 
     this._panelOpen = true;
     this._floatPlaceholder();
+
+    if (!this._unbindGlobalListener) {
+      this._unbindGlobalListener = this._renderer.listenGlobal('document', 'click',
+        (event: MouseEvent) => {
+          if (!this._inputContainer._elementRef.nativeElement.contains(event.target) &&
+              !this._overlayRef.overlayElement.contains(event.target as HTMLElement)) {
+            this._outsideClickStream.next(null);
```

Comment:
IIRC, I didn't go with `fromEvent`, because it may break server-side rendering. Thinking about it again, it should be fine since the menu shows up as a result from a user action.